openedx/core/djangoapps/coursegraph/tasks.py

```diff
@@ -0,0 +1,386 @@
+        transaction.rollback()
+
+
+class ModuleStoreSerializer(object):
+    """
+    Class with functionality to serialize a modulestore into subgraphs,
+    one graph per course.
+    """
+
+    def __init__(self, course_keys):
+        self.course_keys = course_keys
+
+    @classmethod
+    def create(cls, courses=None, skip=None):
+        """
+        Sets the object's course_keys attribute from the `courses` parameter.
+        If that parameter isn't furnished, loads all course_keys from the
+        modulestore.
+        Filters out course_keys in the `skip` parameter, if provided.
+        Args:
+            courses: A list of string serializations of course keys.
+                For example, ["course-v1:org+course+run"].
+            skip: Also a list of string serializations of course keys.
+        """
+        if courses:
+            course_keys = [CourseKey.from_string(course.strip()) for course in courses]
+        else:
+            course_keys = [
+                course.id for course in modulestore().get_course_summaries()
+            ]
+        if skip is not None:
+            skip_keys = [CourseKey.from_string(course.strip()) for course in skip]
+            course_keys = [course_key for course_key in course_keys if course_key not in skip_keys]
+        return cls(course_keys)
+
+    def dump_courses_to_neo4j(self, credentials, override_cache=False):
+        """
+        Method that iterates through a list of courses in a modulestore,
+        serializes them, then submits tasks to write them to neo4j.
+        Arguments:
+            credentials (dict): the necessary credentials to connect
+              to neo4j and create a py2neo `Graph` object
+            override_cache: serialize the courses even if they'be been recently
+                serialized
+
+        Returns: two lists--one of the courses that were successfully written
+            to neo4j and one of courses that were not.
+        """
+
+        total_number_of_courses = len(self.course_keys)
+
+        submitted_courses = []
+        skipped_courses = []
+
+        graph = authenticate_and_create_graph(credentials)
+
+        for index, course_key in enumerate(self.course_keys):
+            # first, clear the request cache to prevent memory leaks
+            RequestCache.clear_request_cache()
+
+            log.info(
+                "Now submitting %s for export to neo4j: course %d of %d total courses",
+                course_key,
+                index + 1,
+                total_number_of_courses,
+            )
+
+            if not (override_cache or should_dump_course(course_key, graph)):
+                log.info("skipping submitting %s, since it hasn't changed", course_key)
+                skipped_courses.append(six.text_type(course_key))
+                continue
+
+            dump_course_to_neo4j.apply_async(
+                args=[six.text_type(course_key), credentials],
+                routing_key=settings.COURSEGRAPH_JOB_QUEUE,
```

Comment:
You can set the routing key either here (at time of calling the task) or include it on the task decorator [like this](https://github.com/edx/edx-platform/blob/9c4869c1d59899a1e4a0ed2bf9cf92c77acc4447/lms/djangoapps/grades/tasks.py#L125) to suggest all callers ought to use said routing key.